From 39723da23f5603ab20710501cf85f3b0654f29d5 Mon Sep 17 00:00:00 2001
From: LE BLOND <erwanleblond@gmail.com>
Date: Mon, 6 Oct 2025 11:31:57 +0200
Subject: [PATCH 1/2] first patch to build it

---
 source/app/ldecod/configfile.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/source/app/ldecod/configfile.h b/source/app/ldecod/configfile.h
index 1e53e9a..326b104 100644
--- a/source/app/ldecod/configfile.h
+++ b/source/app/ldecod/configfile.h
@@ -18,7 +18,7 @@
 //#define LEVEL_IDC       21
 
 
-InputParameters cfgparams;
+extern InputParameters cfgparams;
 
 #ifdef INCLUDED_BY_CONFIGFILE_C
 // Mapping_Map Syntax:
-- 
2.34.1


From dc8795f195327f374bbfc1c7c94f873b5eef74a4 Mon Sep 17 00:00:00 2001
From: LE BLOND <erwanleblond@gmail.com>
Date: Mon, 6 Oct 2025 11:32:28 +0200
Subject: [PATCH 2/2] second patch to build it

---
 source/app/ldecod/defines.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/source/app/ldecod/defines.h b/source/app/ldecod/defines.h
index 6d9835a..5f3573b 100644
--- a/source/app/ldecod/defines.h
+++ b/source/app/ldecod/defines.h
@@ -229,7 +229,7 @@ typedef enum {
 } I8x8PredModes;
 
 // Color components
-enum {
+typedef enum {
   Y_COMP = 0,    // Y Component
   U_COMP = 1,    // U Component
   V_COMP = 2,    // V Component
-- 
2.34.1

