From 03c4d77c8189ee01bf0705a60bf9e361eb37ee48 Mon Sep 17 00:00:00 2001
From: LE BLOND <erwanleblond@gmail.com>
Date: Mon, 6 Oct 2025 14:08:37 +0200
Subject: [PATCH 1/2]  quick fix to build it

---
 source/Lib/DecoderLib/DecLib.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/source/Lib/DecoderLib/DecLib.cpp b/source/Lib/DecoderLib/DecLib.cpp
index 4c06188e0..ead39c9f6 100644
--- a/source/Lib/DecoderLib/DecLib.cpp
+++ b/source/Lib/DecoderLib/DecLib.cpp
@@ -3164,7 +3164,7 @@ void DecLib::xCheckMixedNalUnit(Slice* pcSlice, SPS *sps, InputNALUnit &nalu)
 /**
 - lookahead through next NAL units to determine if current NAL unit is the first NAL unit in a new picture
 */
-bool DecLib::isNewPicture(std::ifstream *bitstreamFile, class InputByteStream *bytestream)
+bool DecLib::isNewPicture(std::istream *bitstreamFile, class InputByteStream *bytestream)
 {
   bool ret = false;
   bool finished = false;
@@ -3273,7 +3273,7 @@ bool DecLib::isNewPicture(std::ifstream *bitstreamFile, class InputByteStream *b
 /**
 - lookahead through next NAL units to determine if current NAL unit is the first NAL unit in a new access unit
 */
-bool DecLib::isNewAccessUnit( bool newPicture, std::ifstream *bitstreamFile, class InputByteStream *bytestream )
+bool DecLib::isNewAccessUnit( bool newPicture, std::istream *bitstreamFile, class InputByteStream *bytestream )
 {
   bool ret = false;
   bool finished = false;
-- 
2.34.1


From ddbb29e29dc9ec24c4f30825b93cf014c084d312 Mon Sep 17 00:00:00 2001
From: LE BLOND <erwanleblond@gmail.com>
Date: Mon, 6 Oct 2025 14:08:54 +0200
Subject: [PATCH 2/2]  quick fix to build it 2

---
 source/Lib/DecoderLib/DecLib.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/source/Lib/DecoderLib/DecLib.h b/source/Lib/DecoderLib/DecLib.h
index 4dc88e65d..89369bdbd 100644
--- a/source/Lib/DecoderLib/DecLib.h
+++ b/source/Lib/DecoderLib/DecLib.h
@@ -277,8 +277,8 @@ public:
   }
 
   void  setAPSMapEnc( ParameterSetMap<APS>* apsMap ) { m_apsMapEnc = apsMap;  }
-  bool  isNewPicture( std::ifstream *bitstreamFile, class InputByteStream *bytestream );
-  bool  isNewAccessUnit( bool newPicture, std::ifstream *bitstreamFile, class InputByteStream *bytestream );
+  bool  isNewPicture( std::istream *bitstreamFile, class InputByteStream *bytestream );
+  bool  isNewAccessUnit( bool newPicture, std::istream *bitstreamFile, class InputByteStream *bytestream );
 
   bool      getHTidExternalSetFlag()               const { return m_mTidExternalSet; }
   void      setHTidExternalSetFlag(bool mTidExternalSet)  { m_mTidExternalSet = mTidExternalSet; }
-- 
2.34.1

